### PR TITLE
Accept .5 for half

### PR DESCRIPTION
**Pull request by som-snytt**
Previous refactor pruned too much by half.

### CODE REVIEW

File: test/junit/scala/collection/StringParsersTest.scala [hunk 1]
```diff
@@ -242,6 +242,7 @@ class StringParsersTest {
     "-0x0.000000000000090000000000000000001p-1022",
     "-0x0.00000000000009fffffffffffffffffffffffffffffffffp-1022",
     "-0x0.0000000000000fffffffffffffffffffffffffffffffffffp-1022",
+    ".4",
```
Comment (martijnhoekstra) [inline test/junit/scala/collection/StringParsersTest.scala]: Can we get "", "." and "4." too while we're at it?

Comment (som-snytt) [reply test/junit/scala/collection/StringParsersTest.scala]: OK, I was too busy with "half" puns.

Comment (martijnhoekstra) [reply test/junit/scala/collection/StringParsersTest.scala]: Maybe change to "0.5" then 👍

**Comment by SethTisue** [comment]
reduce to two commits for merge...?

**Comment by som-snytt** [comment]
I squashed the new commits to .5 or half the total.

**Review by SethTisue — APPROVED** [review]
assuming CI likes it.

to anyone confused about what the context for this, it's https://github.com/scala/scala/pull/6726

**Comment by som-snytt** [comment]
The context is my hasty review of same. I was all ferocious about boolean expressions, irrespective of truth value.

**Comment by som-snytt** [comment]
Woo-hoo, off to rebase all my outstanding PRs! Outstanding in both senses.

**Comment by SethTisue** [comment]
out standing in the rain.

**Pull request merged**